bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonDecoder.kt

```diff
@@ -213,13 +213,37 @@ private class PolymorphicDecoder(
     configuration: BsonConfiguration
 ) : DefaultBsonDecoder(reader, serializersModule, configuration) {
     private var index = 0
+    private var mark: BsonReaderMark? = reader.mark
```

Comment:
Here a mark is set allowing the `PolymorphicDecoder` to iterate to the discriminator field instead of expecting it to be the first field.